kernel/kernel-api/src/test/scala/io/delta/kernel/internal/checksum/ChecksumReadWriteSuite.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.checksum
+
+import io.delta.kernel.data.{ArrayValue, ColumnVector, MapValue, Row}
+import io.delta.kernel.internal.actions.{Format, Metadata, Protocol}
+import io.delta.kernel.internal.data.GenericRow
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.snapshot.SnapshotHint
+import io.delta.kernel.internal.util.InternalUtils.singletonStringColumnVector
+import io.delta.kernel.internal.util.VectorUtils
+import io.delta.kernel.test.{BaseMockJsonHandler, MockEngineUtils}
+import io.delta.kernel.types.{StringType, StructType}
+import io.delta.kernel.utils.CloseableIterator
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.util
+import java.util.{Collections, HashMap, Optional, OptionalLong}
+
+class ChecksumReadWriteSuite extends AnyFunSuite with MockEngineUtils {
+
+  private val FAKE_DELTA_LOG_PATH = new Path("/path/to/delta/log")
+
+
+  test("basic checksum write") {
+    val jsonHandler = new MockCheckSumFileJsonWriter()
+    val checksumWriter =
+      new ChecksumWriter(mockEngine(jsonHandler = jsonHandler), FAKE_DELTA_LOG_PATH)
+    val protocol = createTestProtocol()
+    val metadata = createTestMetadata()
+    val snapshotHint = new SnapshotHint(
+      1,
+      protocol,
+      metadata,
+      OptionalLong.of(100),
+      OptionalLong.of(1))
+    checksumWriter.maybeWriteCheckSum(snapshotHint, "tnx")
+
```

Comment:
validate that no additional rows were written to the file (only 1 row was written and it has the schema we expect)